titus-server-master/src/main/java/com/netflix/titus/master/mesos/DefaultV3TaskInfoFactory.java

```diff
@@ -154,12 +155,20 @@ public DefaultV3TaskInfoFactory(MasterConfiguration masterConfiguration,
         }
         containerInfoBuilder.setKillWaitSeconds(killWaitSeconds);
 
+        // Send passthrough attributes that begin with the agent prefix
+        containerAttributes.forEach((k, v) -> {
+            if (k.startsWith(PASSTHROUGH_ATTRIBUTES_PREFIX)) {
+                containerInfoBuilder.putPassthroughAttributes(k, v);
+            }
+        });
+
         // Configure Environment Variables
-        Map<String, String> userProvidedEnv = container.getEnv().entrySet()
-                .stream()
-                .filter(e -> e.getValue() != null)
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-        containerInfoBuilder.putAllUserProvidedEnv(userProvidedEnv);
+        container.getEnv().forEach((k, v) -> {
+            if (v != null) {
```

Comment:
v != null? and v != ""?

Reply:
This would change the current behavior so we would need to first see if anyone is depending on empty string env variables.